lib/ohai/mixin/which.rb

```diff
@@ -0,0 +1,38 @@
+#--
+# Copyright:: Copyright (c) Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "chef-utils/dsl/which" unless defined?(ChefUtils::DSL::Which)
+require "chef-utils/dsl/default_paths" unless defined?(ChefUtils::DSL::DefaultPaths)
+require_relative "chef_utils_wiring" unless defined?(Ohai::Mixin::ChefUtilsWiring)
+
+module Ohai
+  module Mixin
+    module Which
+      include ChefUtils::DSL::Which
+      include ChefUtils::DSL::DefaultPaths
+      include ChefUtilsWiring
+
+      private
+
+      # we dep-inject default paths into this API for historical reasons
+      #
+      # @api private
+      def __extra_path
```

Comment:
its appropriate for ohai and for chef internally but this is also the which helper that we inject into the cookbook DSL, that comment is aimed at the cookbook usage of it.  it works when paired up with `shell_out` which also has default paths by default, but for the execute resource it would potentially find executables that execute in the chef DSL wouldn't be able to find.  this is just copypasta from:  https://github.com/chef/chef/blob/master/lib/chef/mixin/which.rb#L31-L36